homeassistant/components/wyoming/tts.py

```diff
@@ -0,0 +1,161 @@
+"""Support for Wyoming text to speech services."""
+from collections import defaultdict
+import io
+import logging
+import wave
+
+from wyoming.audio import AudioChunk, AudioChunkConverter, AudioStop
+from wyoming.client import AsyncTcpClient
+from wyoming.tts import Synthesize
+
+from homeassistant.components import tts
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .data import WyomingService
+from .error import WyomingError
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Wyoming speech to text."""
+    service: WyomingService = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        [
+            WyomingTtsProvider(config_entry, service),
+        ]
+    )
+
+
+class WyomingTtsProvider(tts.TextToSpeechEntity):
+    """Wyoming text to speech provider."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        service: WyomingService,
+    ) -> None:
+        """Set up provider."""
+        self.service = service
+        self._tts_service = next(tts for tts in service.info.tts if tts.installed)
+
+        voice_languages: set[str] = set()
+        self._voices: dict[str, list[tts.Voice]] = defaultdict(list)
+        for voice in self._tts_service.voices:
+            if not voice.installed:
+                continue
+
+            voice_languages.update(voice.languages)
+            for language in voice.languages:
+                self._voices[language].append(
+                    tts.Voice(
+                        voice_id=voice.name,
+                        name=voice.name,
+                    )
+                )
+
+        self._supported_languages: list[str] = list(voice_languages)
+
+        self._attr_name = self._tts_service.name
+        self._attr_unique_id = f"{config_entry.entry_id}-tts"
+
+    @property
+    def name(self) -> str | None:
```

Comment:
We don't need the `name` property since we set `_attr_name` above.

Reply:
https://github.com/home-assistant/core/pull/91930